From 40e9ed3d5221cd5cf2ef17744dcf7ab90a1ae020 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Wed, 7 Mar 2018 11:09:41 +0100
Subject: GCC: do not initialize NEON int32x4_t with braces initializer

GCC does not support int32x4_t loading using braces wrapping 4 ints.
Instead we should use the NEON intrinsic to load from an array of
ints.

diff --git a/skia/ext/convolver_neon.cc b/skia/ext/convolver_neon.cc
index 26b91b9c88c4..cae6bc2f833f 100644
--- a/skia/ext/convolver_neon.cc
+++ b/skia/ext/convolver_neon.cc
@@ -23,7 +23,7 @@ AccumRemainder(const unsigned char* pixels_left,
     remainder[2] += coeff * pixels_left[i * 4 + 2];
     remainder[3] += coeff * pixels_left[i * 4 + 3];
   }
-  return {remainder[0], remainder[1], remainder[2], remainder[3]};
+  return vld1q_s32(remainder);
 }
 
 // Convolves horizontally along a single row. The row data is given in
@@ -336,4 +336,4 @@ void ConvolveVertically_Neon(const ConvolutionFilter1D::Fixed* filter_values,
   }
 }
 
-}  // namespace skia
\ No newline at end of file
+}  // namespace skia
-- 
2.14.1

